phoenix-core/src/main/java/org/apache/phoenix/index/IndexMaintainer.java

```diff
@@ -195,7 +207,7 @@ public static void serializeServerMaintainedIndexes(PTable dataTable, ImmutableB
         if (onlyLocalIndexes) {
             if (!dataTable.isTransactional()
                     || !dataTable.getTransactionProvider().getTransactionProvider().isUnsupported(Feature.MAINTAIN_LOCAL_INDEX_ON_SERVER)) {
-                indexesItr = maintainedLocalIndexes(indexes.iterator());
+                indexesItr = maintainedLocalOrGlobalIndexesWithoutMatchingStorageScheme(dataTable, indexes.iterator());
```

Comment:
Shouldn't this still be maintainedLocalIndexes since it is called inside if (onlyLocalIndexex)

Reply:
Unfortunately onlyLocalIndexes variable name is misleading. So it is immutableIndexes or transactional indexes
boolean onlyLocalIndexes = dataTable.isImmutableRows() || dataTable.isTransactional();